<launch>
  <arg name="input_objects" />
  <arg name="input_points_no_ground" />
  <arg name="input_lanelet_map" />
  <arg name="input_scenario" />
  <arg name="output_grid_map" />
  <arg name="output_occupancy_grid" />

  <node pkg="costmap_generator" type="costmap_generator" name="costmap_generator" output="screen">
    <remap from="~input/objects" to="$(arg input_objects)" />
    <remap from="~input/points_no_ground" to="$(arg input_points_no_ground)" />
    <remap from="~input/vector_map" to="$(arg input_lanelet_map)" />
    <remap from="~input/scenario" to="$(arg input_scenario)" />
    <remap from="~output/grid_map" to="$(arg output_grid_map)" />
    <remap from="~output/occupancy_grid" to="$(arg output_occupancy_grid)" />

    <param name="costmap_frame" value="map" />
    <param name="vehicle_frame" value="base_link" />
    <param name="map_frame" value="map" />

    <param name="update_rate" value="10.0" />

    <param name="use_wayarea" value="true" />
    <param name="use_objects" value="true" />
    <param name="use_points" value="true" />

    <param name="grid_min_value" value="0.0" />
    <param name="grid_max_value" value="1.0" />
    <param name="grid_resolution" value="0.2" />
    <param name="grid_length_x" value="70" />
    <param name="grid_length_y" value="70" />
    <param name="grid_position_x" value="0" />
    <param name="grid_position_y" value="0" />
    <param name="maximum_lidar_height_thres" value="0.3" />
    <param name="minimum_lidar_height_thres" value="-2.2" />
    <param name="expand_polygon_size" value="1.0" />
    <param name="size_of_expansion_kernel" value="9" />
  </node>
</launch>
